torch/testing/_asserts.py

```diff
@@ -577,31 +577,43 @@ def _to_tensor(array_or_scalar_like: Any) -> Tuple[Optional[_TestingErrorMeta],
     return None, tensor
 
 
-def _to_tensor_pair(actual: Any, expected: Any) -> Tuple[Optional[_TestingErrorMeta], Optional[_TensorPair]]:
+def _check_types(actual: Any, expected: Any, *, allow_subclasses: bool) -> Optional[_TestingErrorMeta]:
+    # We exclude numbers here, since numbers of different type, e.g. int vs. float, should be treated the same as
+    # tensors with different dtypes. Without user input, passing numbers of different types will still fail, but this
+    # can be disabled by setting `check_dtype=False`.
+    if isinstance(actual, numbers.Number) and isinstance(expected, numbers.Number):
+        return None
+
+    msg_fmtstr = f"Except for Python scalars, {{}}, but got {type(actual)} and {type(expected)} instead."
+    common_superclass = isinstance(actual, type(expected)) or isinstance(expected, type(actual))
```

Comment:
It's not that the object share a superclass, but that one object has the other's class as a superclass.

Reply:
Also: let's add a comment here explaining how isinstance works with sub/super classes.

Reply:
> Also: let's add a comment here explaining how isinstance works with sub/super classes.

Not sure I get the point? If one knows what `isinstance` does, the implementation is straight forward. If not one should read the documentation?

Reply:
> It's not that the object share a superclass, but that one object has the other's class as a superclass.

I went with the "directly related" terminology, because "one object has the other's class as a superclass." is also not completely right: both objects can have the same time (arguably the more common case) and this check would still pass.